src/helpers/parser.ts

```diff
@@ -25,19 +29,52 @@ export const gitIssueParser = async ({ owner, repo, issue_number }: GitParser):
   }
 };
 
-export const gitLinkedIssueParser = async ({ owner, repo, issue_number }: GitParser): Promise<string> => {
+export const gitLinkedIssueParser = async ({ owner, repo, pull_number }: GitParser): Promise<string> => {
   try {
-    const { data } = await axios.get(`https://github.com/${owner}/${repo}/pull/${issue_number}`);
+    const { data } = await axios.get(`https://github.com/${owner}/${repo}/pull/${pull_number}`);
     const dom = parse(data);
     const devForm = dom.querySelector("[data-target='create-branch.developmentForm']") as HTMLElement;
-    const linkedPRs = devForm.querySelectorAll(".my-1");
+    const linkedIssues = devForm.querySelectorAll(".my-1");
 
-    if (linkedPRs.length === 0) {
+    if (linkedIssues.length === 0) {
       return "";
     }
 
-    const prUrl = linkedPRs[0].querySelector("a")?.attrs?.href || "";
-    return prUrl;
+    const issueUrl = linkedIssues[0].querySelector("a")?.attrs?.href || "";
+    return issueUrl;
+  } catch (error) {
+    return "";
+  }
+};
+
+export const gitLinkedPrParser = async ({ owner, repo, issue_number }: GitParser) => {
+  try {
+    const { data } = await axios.get(`https://github.com/${owner}/${repo}/issues/${issue_number}`);
+    const context = getBotContext();
+    const payload = context.payload as Payload;
+    const dom = parse(data);
+    const devForm = dom.querySelector("[data-target='create-branch.developmentForm']") as HTMLElement;
+    const linkedPRs = devForm.querySelectorAll(".my-1");
+    if (linkedPRs.length === 0) return "";
+    let linkedPullRequest;
+    for (const linkedPr of linkedPRs) {
+      const prHref = linkedPr.querySelector("a")?.attrs?.href || "";
+      const parts = prHref.split("/");
+      // extract the organization name and repo name from the link:(e.g. "https://github.com/wannacfuture/ubiquibot/pull/5";)
+      const organization = parts[parts.length - 4];
+      const repository = parts[parts.length - 3];
+
+      if (`${organization}/${repository}` !== payload.repository.full_name) continue;
+      const prNumber = prHref.substring(prHref.lastIndexOf("/") + 1);
+      const pr = await getPullByNumber(context, Number(prNumber));
+      if (!pr || !pr.merged) continue;
+
+      if (!linkedPullRequest) linkedPullRequest = pr;
+      else if (linkedPullRequest.merged_at && pr.merged_at && new Date(linkedPullRequest.merged_at) < new Date(pr.merged_at)) {
+        linkedPullRequest = pr;
+      }
+    }
+    return linkedPullRequest;
   } catch (error) {
     return "";
```

Comment:
Shouldn't this be something that clearly expresses an issue? Like `false` or `null`?

Reply:
if will be `undefined` when there is no linked pr.